c/meterpreter/source/extensions/stdapi/server/fs/search.c

```diff
@@ -757,10 +756,10 @@ DWORD search(WDS_INTERFACE * pWDSInterface, wchar_t *directory, SEARCH_OPTIONS *
 	return dwResult;
 }
 
-VOID search_all_drives(WDS_INTERFACE *pWDSInterface, SEARCH_OPTIONS *options, Packet *pResponse)
+DWORD search_all_drives(WDS_INTERFACE *pWDSInterface, SEARCH_OPTIONS *options, Packet *pResponse) //!!! VOID -> DWORD
 {
 	DWORD dwLogicalDrives = GetLogicalDrives();
-
+	DWORD dwResult = ERROR_ACCESS_DENIED;

```

Comment:
Are we sure that ERROR_ACCESS_DENIED is the only possible error if we fail to move further?  Can we get that error to return from an action rather than assuming that was the problem?  I see this is what others did before, but still makes me wonder if there would be a more definitive way.

Reply:
As you mentioned, I used the kind of construction based on the function "search". Actually, it is not necessary to specify variable dwResult, because subfunctions will change its' value further.